user/TASK/gimbal_task/gimbal_task.c

```diff
@@ -26,73 +26,119 @@
 #include "remote_control.h"
 #include "USART_comms.h"
 #include <stdio.h>
+#include "pid.h"
 
-float rc_channel_1;
-float rc_channel_2;
-const RC_ctrl_t* rc_ptr;
+#define DEADBAND 10
+
+// This is accessbile globally and some data is loaded from INS_task
 Gimbal_t gimbal;
 
 //UART mailbox
 char str[32] = {0};
 
+/******************** Functions ********************/
+static void initialization(Gimbal_t *gimbal);
+static void get_new_data(Gimbal_t *gimbal);
+static void update_setpoints(Gimbal_t *gimbal);
+static void increment_PID(Gimbal_t *gimbal);
+
 
 /**
- * @brief Initialises PID and fetches Gimbal motor data to ensure 
+ * @brief Initializes PID and fetches Gimbal motor data to ensure 
  *  gimbal points in direction specified. 
  * @param FreeRTOS parameters
  * @retval None
  */
 void gimbal_task(void* parameters){
 
-    vTaskDelay(GIMBAL_TASK_INIT_TIME);
-	
-    gimbal.pitch_motor->gimbal_motor_raw = get_Pitch_Gimbal_Motor_Measure_Point();
-    gimbal.yaw_motor->gimbal_motor_raw = get_Yaw_Gimbal_Motor_Measure_Point();
-
-    fp32 pid_constants[3] = {pid_kp, pid_ki, pid_kd};
-    PidTypeDef pid_pitch;
-    PID_Init(&pid_pitch, PID_POSITION, pid_constants, max_out, max_iout);
-
-    PidTypeDef pid_yaw;
-    PID_Init(&pid_yaw, PID_POSITION, pid_constants, max_out, max_iout);
-    
-
-    rc_ptr = get_remote_control_point();
+    vTaskDelay(200);
```

Comment:
I see this value is defined in gimbal_task.h as 300... Hmm

Reply:
Updated to 300 + define